snippets/firestore-next/test-firestore/set_with_merge.js

```diff
@@ -7,6 +7,6 @@
 // [START set_with_merge_modular]
 import { doc, setDoc } from "firebase/firestore"; 
 
-const cityRef = doc(db, 'cities', 'BJ');
+const cityRef = await doc(db, 'cities', 'BJ');
 setDoc(cityRef, { capital: true }, { merge: true });
```

Comment:
```suggestion
const cityRef = doc(db, 'cities', 'BJ');
await setDoc(cityRef, { capital: true }, { merge: true });
```

`doc` [doesn't return a Promise](https://firebase.google.com/docs/reference/js/firestore_#doc), but [`setDoc`  does](https://firebase.google.com/docs/reference/js/firestore_#setdoc).

Reply:
Yep!